### DISCUSSION TITLE
`now dev` local development not parsing req.body from external domain

### DISCUSSION THREAD

**Discussion by benoror** [vercel/vercel#4218] [General] [answered]
Hi! My use-case involves doing local development while getting requests from an external service into my `localhost`.

```javascript
module.exports = async (req, res) => {
  console.log(req.body)
  res.status(200).json(req.body)
}

```

It works when I locally `curl -X POST localhost:3000/api -H "Content-Type: application/json" -d '{"foo":"bar"}'  `

But when getting the request from an external host `req.body` is `undefined`

**Comment by benoror** [comment]
I tried setting [`now.json` headers](https://vercel.com/docs/v2/edge-network/headers#request-headers) to allow any origin, but I had no success:

```json
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        {
          "key" : "Access-Control-Allow-Origin",
          "value" : "/*"
        },
        {
          "key" : "Access-Control-Allow-Headers",
          "value" : "Origin, X-Requested-With, Content-Type, Accept"
        },
        {
          "key" : "Access-Control-Allow-Credentials",
          "value" : "true"
        }
      ]
    }
  ]

```

**Comment by coetry** [comment, marked as answer]
@benoror External sources have no way to reach your local domain unless you open up a tunnel to your local machine with something like [ngrok](https://ngrok.com/).

**Reply by benoror** [reply]
I forgot to mention I am indeed using `ngrok` to create a tunnel
